apollo-router/src/spec/query.rs

```diff
@@ -46,7 +45,7 @@ pub(crate) struct Query {
     string: String,
     #[derivative(PartialEq = "ignore", Hash = "ignore", Debug = "ignore")]
     #[serde(skip)]
-    compiler: OnceCell<Mutex<ApolloCompiler>>,
+    compiler: Mutex<ApolloCompiler>,
```

Comment:
This makes deserialization from Redis cache rely on `impl Default for ApolloCompiler` which incorrectly gives an empty compiler. The role of `OnceCell` here was not just about async but also to let us initialize the compiler from `&Schema` and `self.string`. I think we can still replace `OnceCell` with `Option::get_or_insert_with` by storing `Mutex<Option<ApolloCompiler>>`, since the mutex already provides synchronization.

Even though we don’t have anything yet relying on the compiler, when we do we’ll definitely want it to be schema-aware.

Reply:
good point, thanks for calling it out